src/csharp/Grpc.Core.Tests/Internal/AsyncCallTest.cs

```diff
@@ -493,6 +493,20 @@ public void ServerStreaming_StartCallFailureDoesntLeakResources()
             Assert.IsTrue(fakeCall.IsDisposed);
         }
 
+        [Test]
+        public void ServerStreaming_ServerResponseFailureDoesntLeakResources()
```

Comment:
nit: rename to 
`ServerStreaming_OnReceivedResponseHeadersFailureDoesntLeakResources()` to be more specific?